src/test/scala/com/fulcrumgenomics/util/RefFlatSourceTest.scala

```diff
@@ -0,0 +1,129 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package com.fulcrumgenomics.util
+
+import java.nio.file.Paths
+
+import com.fulcrumgenomics.testing.UnitSpec
+import com.fulcrumgenomics.util.GeneAnnotations.Exon
+import htsjdk.samtools.{SAMSequenceDictionary, SAMSequenceRecord}
+
+class RefFlatSourceTest extends UnitSpec {
+
+  // Adapted from http://hgdownload.cse.ucsc.edu/goldenPath/hg18/database/refFlat.txt.gz
+  private val RefFlatFile = Paths.get("src/test/resources/com/fulcrumgenomics/util/refFlat.txt.gz")
+
+  private val Dictionary = {
+    import scala.collection.JavaConversions.seqAsJavaList
+    new SAMSequenceDictionary(Seq(new SAMSequenceRecord("chr1", 249250621)))
+  }
+
+  "RefFlatSource" should "read valid refFlat from various kinds of input resources" in {
+    val rf = RefFlatFile
+    val lines = Io.readLines(rf)
+
+
+
+    Seq(
+      RefFlatSource(rf, Some(Dictionary)),
+      RefFlatSource(rf.toFile, Some(Dictionary)),
+      RefFlatSource(Io.toInputStream(rf), Some(Dictionary)),
+      RefFlatSource(lines, Some(Dictionary))
+    ).foreach(source => {
+      val genes = source.toSeq
+
+      // # of genes
+      genes.length shouldBe 142
+
+      // # of transcripts across all genes
+      genes.map(_.size).sum shouldBe 360
+
+      // # of exons across all transcripts across all genes
+      genes.map { gene => gene.map(_.size).sum }.sum shouldBe 4613
+
+      // verify the first gene
+      val gene = genes.head
+      gene.contig shouldBe "chr1"
+      gene.start shouldBe 141638944
+      gene.end shouldBe 141655128
+      gene.negativeStrand shouldBe true
+      gene.name shouldBe "ANKRD20A12P"
+      gene.size shouldBe 1
+
+      // verify the first transcript in the first gene
+      val transcript = gene.head
+      transcript.name shouldBe "NR_046228"
+      transcript.txStart shouldBe 141638944
+      transcript.txEnd shouldBe 141655128
+      transcript.cdsStart shouldBe 141655129
+      transcript.cdsEnd shouldBe 141655128
+      transcript.size shouldBe 5
+
+      //verify the first exon in the first transcript
+      val exon = transcript.head
+      exon shouldBe Exon(141638944, 141639396)
+    })
+  }
+
+  it should "fail if the chromosomes differ across transcripts" in {
```

Comment:
Harping on my earlier point, this is not uncommon.  Running the following on the hg38 refFlat file from UCSC:
```bash
gzcat refFlat.txt.gz | cut -f1,3,4 | sort -u | cut -f1 | sort | uniq -c | sort -rn | \
    awk '{if ($1 > 1) dupes +=1; total+=1;} END {print dupes, total, dupes/total}'
```
yeilds:
`1491 27238 0.0547397`

I.e. there are 1491 genes that have more than one `contig,strand` in the file.  Even if you exclude mappings to `alt` and `random` contigs it's still:
`442 27221 0.0162375`